[Bug] Test command does not fail even if tests fail

**Describe the bug**
When running `npm run test`, if some mocha tests fail, the command's exit code is 0.

**To Reproduce**
Steps to reproduce the behavior:
1. Run `npm run test`
2. Observe that there are no npm error logs, and the command's exit code is 0 

**Expected behavior**
The command's exit code should be 1 or a non 0 number since the tests fail.

**Screenshots**
When the tests fail, we should see these logs like these at the end

> npm ERR! code ELIFECYCLE
npm ERR! errno 2
npm ERR! porn-vault@1.0.0 test: `mocha -r ts-node/register test/**/*.spec.ts`
npm ERR! Exit status 2
npm ERR! 
npm ERR! Failed at the porn-vault@1.0.0 test script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
npm ERR! A complete log of this run can be found in:
npm ERR!     /home/chris/.npm/_logs/2020-05-17T10_56_09_624Z-debug.log

**Desktop (please complete the following information):**
 - OS: Ubuntu 18.04

**Additional context**
It seems like any test file that imports the `src/index.ts` file (even indirectly) provokes the command to not fail.
But this is difficult to test, since so many modules have indirect dependencies, I don't even know how to find the root issue.

--- a/.mocharc.js
+++ b/.mocharc.js
@@ -1,1 +1,2 @@
 process.env.PREVENT_STARTUP = "true";
+process.env.NODE_ENV = "test";

--- a/src/index.ts
+++ b/src/index.ts
@@ -23,7 +23,13 @@ function killProcess(code = 0) {
       logger.log("Killing izzy...");
       izzyProcess.kill();
     }
-    process.exit(code);
+
+    // When running tests, we want to be able to cleanup any services,
+    // but we cannot overload the actual 'exit' otherwise mocha's
+    // exit code will not reflect the actual result of the tests
+    if (process.env.NODE_ENV !== "test") {
+      process.exit(code);
+    }
   };
 }
 

